rts/motoko-rts/src/gc/mark_compact.rs

```diff
@@ -108,11 +104,12 @@ unsafe fn mark_static_roots<M: Memory>(mem: &mut M, static_roots: SkewedPtr, hea
     // Static objects are not in the dynamic heap so don't need marking.
     for i in 0..root_array.len() {
         let obj = root_array.get(i).unskew() as *mut Obj;
-        mark_fields(mem, obj, heap_base);
+        mark_fields(mem, obj, obj.tag(), heap_base);
     }
 }
 
 unsafe fn push_mark_stack<M: Memory>(mem: &mut M, obj: SkewedPtr, heap_base: u32) {
```

Comment:
maybe mark_object would be a better name?

Reply:
Done in #2648.